Doc/library/multiprocessing.shared_memory.rst

```diff
@@ -64,26 +64,34 @@ copying of data.
    memory block may be larger or equal to the size requested.  When attaching
    to an existing shared memory block, the ``size`` parameter is ignored.
 
+   *track*, when enabled, registers the shared memory block with the resource
+   tracker process.  This process ensures proper cleanup of shared memory
+   blocks even when all other processes with access to the memory have failed
+   to do so (mainly due to being killed by signals).  The resource tracker is
+   overzealous in certain situations and might delete a shared memory block
+   when any process with access to the shared memory has terminated.  *track*
```

Comment:
Indeed, this should qualified more precisely. AFAIU, the "overzealous" part is when some process _unrelated to the current process_ is also using the shared memory.

However, the resource tracker works fine (and as intended) as long as all uses of the shared memory segment have been initiated from the process that created them (for example by passing the segment to tasks running in child processes).

Reply:
@pitrou Would it be fair to say that the resource tracker is overzealous in any situation other than the case where all file descriptors have been inherited (as done by multiprocessing)? I hope I understand this all correctly.

Reply:
Well, it seems that pickling and unpickling a `SharedMemory` object would do the right wrt. the release tracker:
https://github.com/python/cpython/blob/e990e41823981c3993c320492f76a0872db08cf9/Lib/multiprocessing/shared_memory.py#L192-L200

And pickling is how multiprocessing passes most data to related processes (such as tasks, process arguments...) except when using the "fork" method.

Reply:
Thanks, that clears up quite a bit. Would it be precise to say that the issue is due to duplicate resource trackers which multiprocessing somehow manages to avoid?

Reply:
IMHO, the doc should explain in which _context_ that happens, not the underlying cause.